### PR TITLE
Refactor of the code

### PR DESCRIPTION
**Pull request by stebranchi**
I think this could be a slightly better code

### CODE REVIEW

File: jobs/serializers.py [hunk 1]
```diff
@@ -11,14 +11,4 @@ class JobSerializer(serializers.ModelSerializer):
...
-class JobRunSerializer(serializers.ModelSerializer):
-    config = serializers.JSONField()
-    result = serializers.JSONField(required=False)
-    log = LogSerializer(read_only=True)
-
-    class Meta:
-        model = JobRun
-        fields = ('id', 'created_date', 'modified_date', 'config', 'status', 'result', 'type', 'log', 'error')
-
+        fields = ('id', 'created_date', 'modified_date', 'config', 'status', 'result', 'type', 'split', 'error')
```
Comment (TKasekamp) [inline jobs/serializers.py]: You added `run`, don't you want it to be in the JSON?

File: jobs/views.py [hunk 2]
```diff
@@ -120,18 +120,18 @@ def get_prediction(request, pk1, pk2):
                        'prefix_length':1,
                        "rule": "remaining_time",
                        'threshold': 'default',
+                       'log_id':3,
                        }
-    log = Log.objects.get(pk=3)
-    jobrun=JobRun.objects.create(config=config, log=log, type=NEXT_ACTIVITY)
+    jobrun=Job.objects.create(config=config, run=True, type=NEXT_ACTIVITY)
```
Comment (TKasekamp) [inline jobs/views.py]: This method looks like a hack. Update the description to say that this is a demo thing. Or add a TOD

File: jobs/tasks.py [hunk 3]
```diff
@@ -28,14 +31,15 @@ def prediction(job,model):
         raise e
     finally:
         job.save()
-
+        
 def calculate(job,model):
     """ Main entry method for calculations"""
     #print("Start job {} with {}".format(job['type'], get_run(job)))
```
Comment (TKasekamp) [inline jobs/tasks.py]: Did you comment out these print loggers? I'd rather like them to be working, same in `training`.

File: jobs/migrations/0004_auto_20180326_1352.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.0.3 on 2018-03-26 13:52
+
```
Comment (TKasekamp) [inline jobs/migrations/0004_auto_20180326_1352.py]: You can squash multiple migrations to one in PRs with https://docs.djangoproject.com/en/1.11/ref/django-admin/#django-admin-squashmigrations

Not that important, but makes the code cleaner.

File: core/regression.py [hunk 5]
```diff
@@ -112,14 +111,3 @@ def prep_data(test_df):
     test_data = test_data.drop(['trace_id', 'remaining_time'], 1)
 
     return test_data, original_test_data
-
-
-def __choose_regressor(regression_type: str):
```
Comment (TKasekamp) [inline core/regression.py]: Where is the regressor chosen if this is deleted?

File: core/regression.py [hunk 6]
```diff
@@ -65,6 +47,23 @@ def kmeans_run(run_df, regressor, estimator):
...
+    if job['clustering'] == KMEANS:
+        results_df = kmeans_clustering(original_test_data, regressor, estimator)
+    else:
+        results_df = no_clustering(original_test_data, test_data, regressor)
+
+    results = prepare_results(results_df)
+    return results
+
 def kmeans_clustering(original_test_data, regressor, estimator):
     print(estimator)
```
Comment (TKasekamp) [inline core/regression.py]: I don't like simple prints in the completed code. It makes it hard to see what is going on. Fine for local development, but remove them for PRs.

**Comment by TKasekamp** [comment]
Sorry, I didn't notice you had made a PR. You should also remove unused imports, there should be a plugin for Eclipse for that.

**Comment by TKasekamp** [comment]
A general suggestion: If you make a new method, you could add a Python doc comment under it to say that it was made by you. At the moment it's difficult for me to say whats mine and whats yours, especially for stuff in `core`.

**Comment by stebranchi** [comment]
In these days I've already fixed some of these problems, for the choose regressor it's not needed in core because the model is created in training and core module simply takes the model from the database.
Now i'm trying to squash the migrations.
I discovered that i need to change the predmodel model because it's not good right now, so I think that it's better to wait to merge this PR. When i'll finish with predmodels I'll let you know.
For the comment in new method i'll do from now on, by the way I think I have changed like, everything.

**Comment by TKasekamp** [comment]
I can't deploy the back-end to UT servers because the master version is broken. Please do a check of the end-to-end training tasks and fix it. 
```
11:54:02 TypeError: prediction() missing 1 required positional argument: 'model'
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/rq/worker.py", line 789, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 573, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 579, in _execute
    return self.func(*self.args, **self.kwargs)
TypeError: prediction() missing 1 required positional argument: 'model'
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/rq/worker.py", line 789, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 573, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 579, in _execute
    return self.func(*self.args, **self.kwargs)
TypeError: prediction() missing 1 required positional argument: 'model'
```

**Comment by stebranchi** [comment]
Try now if it works fine or not and let me know

**Comment by TKasekamp** [comment]
I don't have the time at the moment to check out your repo and install everything. Frankly, you caused the bug and you have to fix it. 
Start the webserver and start the rqworker. Use the `curl` in the README to see if the task is working correctly. If you don't know how to use that then start the front-end and use the some random training options.

**Comment by TKasekamp** [comment]
I have added you as collaborator to this and the front-end project. For development I'd like you to make a branch and then create a PR so I can see it. This is so we don't have a million repositories with the same code. And all the automated testing is on this repo.

Please don't directly commit stuff to the master branch 😄 

At the end, when you have to submit your thesis, you can make a fork.

**Comment by stebranchi** [comment]
Wait a moment. I didn't ask you to solve the problem I caused , but simply have a feedback if I fixed it right or not, by the way I think I fixed it, because It works fine on my computer. Sorry if my previous message wasn't clear. 
At the moment i'm rewriting the code to work fine wih the new structure of PredModel

**Comment by TKasekamp** [comment]
Ok sorry, if it works then I'd like to merge it so I could deploy it today. Or do you have any more refactoring you want to do here?

**Comment by stebranchi** [comment]
I think that for now you can merge it.

**Pull request merged**